lib/services/mail_service.dart

```diff
@@ -0,0 +1,98 @@
+import 'dart:convert';
+import 'dart:io';
+import 'dart:typed_data';
+
+import 'package:enough_mail/enough_mail.dart';
+import 'package:summer2022/exceptions/fetch_mail_exception.dart';
+import 'package:summer2022/models/Digest.dart';
+
+/// todo: remove this once mailpiece class is properly implemented
+class MailPiece {
+  String ID;
+  String EmailID;
+  DateTime TimeStamp = new DateTime(2022, 1, 1);
+  String Sender;
+  String MidID;
+  String ImageText;
+
+  MailPiece(this.ID, this.EmailID, this.MidID, this.ImageText, this.Sender,
+      this.TimeStamp);
+
+  factory MailPiece.fromJson(dynamic json) {
+    return MailPiece(
+        json['ID'] as String,
+        json['EmailID'] as String,
+        json['MidID'] as String,
+        json['ImageText'] as String,
+        json['Sender'] as String,
+        json['TimeStamp'] as DateTime);
+  }
+}
+
+class MailService {
+  /// Location of mail data file
+  String mailData = "";
+
+  Future<File> get _localFile async {
+    return File(mailData);
+  }
+
+  /// Retrieves all mail from local cache that matches [keyword] or is within [startDate] and [endDate]
+  /// [startDate] and [endDate] should either both have values or both be null
+  /// throws a [FetchMailException] error if file retrieval, parsing, or filtering fails
+  Future<List> fetchMail(
+      String? keyword, DateTime? startDate, DateTime? endDate) async {
+    try {
+      final jsonMail = await _localFile;
```

Comment:
See other comment. I don't think `File(...)` is async.
See: https://api.dart.dev/stable/2.18.1/dart-io/File-class.html